### ISSUE TITLE
bug: DynamoDB delete_expired_items not working properly "Attribute name is a reserved keyword"

### ISSUE THREAD

**Issue by cgaube** [localstack/localstack#11186]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When using the delete expired items endpoint `curl -X DELETE localhost:4566/_aws/dynamodb/expired`
and the function https://github.com/localstack/localstack/blob/b58b0f8ef6b82d38bb847b843e111e98d6c40acf/localstack-core/localstack/services/dynamodb/provider.py#L430 is triggered an error appear

```
 An error occurred when deleting expired items from table XXXXX: An error occurred (ValidationException) when calling the Scan operation: Invalid FilterExpression: Attribute name is a reserved keyword; reserved keyword: TTL
```

The endpoint returns a 200 with `{"ExpiredItems": 0}`   which is invalid

The issue I believe is because the scan function is improperly generated 
it should probably use the `#ttl` notation

`FilterExpression="#ttl <= :threshold",`

instead of `FilterExpression=f"{attribute_name} <= :threshold",` 

### Expected Behavior

Expired items are deleted from dynamo db

### How are you starting LocalStack?

With a docker-compose file

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)

    docker run localstack/localstack

#### Client commands (e.g., AWS SDK code snippet, or sequence of "awslocal" commands)
```
awslocal dynamodb create-table \
    --table-name example-table \
    --attribute-definitions \
        AttributeName=id,AttributeType=S \
        AttributeName=ttl,AttributeType=N \
    --key-schema \
        AttributeName=id,KeyType=HASH \
    --billing-mode PAY_PER_REQUEST \
    --tags Key=Name,Value=example-table

awslocal dynamodb update-time-to-live \
    --table-name example-table \
    --time-to-live-specification Enabled=true,AttributeName=ttl

# Add items
# Add the expired item to the DynamoDB table
awslocal dynamodb put-item \
    --table-name example-table \
    --item '{
        "id": {"S": "item1"},
        "ttl": {"N": "1720713091"}
    }'

curl -X DELETE localhost:4566/_aws/dynamodb/expired
```

### Environment

```markdown
- OS:
- LocalStack:
  LocalStack version:
  LocalStack Docker image sha:
  LocalStack build date:
  LocalStack build git hash:
```

**Comment by giograno** [comment]
Hi @cgaube, thanks for reporting this issue. We have released a fix in the `latest` Docker image (tagged 3.6 also contains the fix). Please let us know if you experience additional issues.

**Comment by cgaube** [comment]
Thank you very much

I have pulled the latest docker image and tested that it works properly 
![Screenshot 2024-07-25 at 08 24 45](https://github.com/user-attachments/assets/3ef30042-9203-4050-890b-2cf16b19775a)
